### ISSUE TITLE
newbie problem cant get fox cloud to work with ha 2022.11.3

### ISSUE THREAD

**Issue by Thebig24fan** [macxq/foxess-ha#93]
hi not been using ha long cant seem to get foxess  cloud to work ive added the new  line of code that was posted to fix 10+ issue and still cant it to work  please help thanks  i get this error 

2022-11-17 17:41:31.254 DEBUG (MainThread) [custom_components.foxess.sensor] Updating data from https://www.foxesscloud.com/
2022-11-17 17:41:31.254 DEBUG (MainThread) [custom_components.foxess.sensor] Token is empty, authenticating for the firts time
2022-11-17 17:41:34.933 ERROR (MainThread) [custom_components.foxess.sensor] Error fetching FoxESS data: Error communicating with API: {"errno":41203,"result":null}
2022-11-17 17:41:34.938 DEBUG (MainThread) [custom_components.foxess.sensor] Finished fetching FoxESS data in 3.683 seconds (success: False)
2022-11-17 17:41:34.938 ERROR (MainThread) [custom_components.foxess.sensor] FoxESS Cloud initializaction failed, fix error and restar ha

thanks for any help you provide

**Comment by robingemmill** [comment]
I think foxess cloud is currently unavailable - my HA data is also available. The error code in your log above ("errno":41203) is the same "Operation Timed Out" error I'm seeing when I try to log on to Foxess Cloud on the pc. I think we might just have to be patient.

**Comment by robingemmill** [comment]
You should be able to log on to Foxess Cloud now - so if it still doesn't work it may be the issue identified in https://github.com/macxq/foxess-ha/issues/83#issue-1435296098. I tried the 2022.11 initially and faced the same problem, so I down graded to 2022.10.5 which is where I still am. Don't think I will update to 11.x until someone with the FoxEss Cloud API installation has confirmed it works ;-)

**Comment by cosmico89** [comment]
I can tell you that today I had the same problem I solved with:
1. downgrade with `core update --version=2022.10.5`
2. In the file `custom_components/foxess/sensor.py` to the line `#177` comment it out and replace with 
`payloadAuth = f'user={username}&password={hashedPassword}'`
3. Update to version `2022.11.3`
4. Restart HA `Settings -> System -> Restart (top right)`

**Comment by DocWoodward** [comment]
2022.11.4 seems to have broken this fix. I have rolled back to 2022.11.3 and everything is working again. Can someone confirm please?

**Comment by cosmico89** [comment]
I have no problems with `2022.11.4`, but in these days the `FoxEss cloud` has had problems, maybe you have had bad luck, in case try restarting everything completely and see if it goes or not from the logs:
`Settting -> System -> Logs`

**Comment by prezes210** [comment]
@DocWoodward "fix" from the above command seems to work fine `2022.11.4` as well

**Comment by DocWoodward** [comment]
Thanks @prezes210

**Comment by prezes210** [comment]
Today stopped working in my instance as well. 
I have the impression that guys are working hard on `foxess` site in last days.
```
2022-11-28 12:34:25.510 ERROR (MainThread) [custom_components.foxess.sensor] Unexpected error fetching FoxESS data: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
self.data = await self._async_update_data()
File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in _async_update_data
return await self.update_method()
File "/config/custom_components/foxess/sensor.py", line 123, in async_update_data
token = await authAndgetToken(hass, username, hashedPassword)
File "/config/custom_components/foxess/sensor.py", line 201, in authAndgetToken
response = json.loads(restAuth.data)
File "/usr/local/lib/python3.10/json/__init__.py", line 346, in loads
return _default_decoder.decode(s)
File "/usr/local/lib/python3.10/json/decoder.py", line 337, in decode
obj, end = self.raw_decode(s, idx=_w(s, 0).end())
File "/usr/local/lib/python3.10/json/decoder.py", line 355, in raw_decode
raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2022-11-28 12:34:25.513 DEBUG (MainThread) [custom_components.foxess.sensor] Finished fetching FoxESS data in 7.602 seconds (success: False)
```

**Comment by ohnsorgesebastian** [comment]
Can anyone help:

Logger: homeassistant.loader
Source: loader.py:745
First occurred: 23:17:07 (1 occurrences)
Last logged: 23:17:07

Unexpected exception importing platform custom_components.foxess.sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 728, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 745, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1017, in get_code
  File "<frozen importlib._bootstrap_external>", line 947, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/config/custom_components/foxess/sensor.py", line 179
    user_agent = user_agent_rotator.get_random_user_agent() 
                                                            ^
IndentationError: unindent does not match any outer indentation level

**Comment by ogmm** [comment]
will someone make an Pull Request with this change?

**Comment by macxq** [comment]
fixed in v0.16  thanks to @guba91
